File: movie-infrastructures/storage/src/main/java/com/movie/storage/movie/entity/ScreenEntity.java
package com.movie.storage.movie.entity;

import com.movie.storage.BaseEntity;
import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.Objects;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Table(name = "screen")
public class ScreenEntity extends BaseEntity {

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(columnDefinition = "BIGINT UNSIGNED")
    private Long id;

    private Long theaterId;

    @Column(columnDefinition = "VARCHAR(10) NOT NULL COMMENT '싱영관 명'")
    private String name;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof ScreenEntity screenEntity)) return false;
        return id != null && id.equals(screenEntity.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }

}

File: infrastructure/src/main/kotlin/io/github/jehwanyoo/redis_1st/entity/ScreenEntity.kt
package io.github.jehwanyoo.redis_1st.entity

import io.github.jehwanyoo.redis_1st.model.Screen
import jakarta.persistence.*

@Entity
@Table(name = "screen")
class ScreenEntity(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long, // 고유 ID

    @Column(nullable = false)
    val name: String, // 상영관 이름

    @Column(name = "seat_row", nullable = false)
    val row: Int, // 좌석 행 수

    @Column(name = "seat_column", nullable = false)
    val column: Int // 좌석 열 수
) {
    fun toDomain(): Screen = Screen(
        id = id,
        name = name,
        row = row,
        column = column
    )
}